### ISSUE TITLE
Accessibility: Make the individual chat list more accessible.

### ISSUE THREAD

**Issue by MarcoZehe** [deltachat/deltachat-desktop#2141]
- Operating System (Linux/Mac/Windows/iOS/Android): All desktop
- Delta Chat Version: 1.14
- Expected behavior: The context menu should come up not only on links, but on all chat items, not just by right-clicking.
- Actual behavior: Context menus only work for focusable elements within each message, if any, like links.
- Steps to reproduce the problem: Open a chat in DeltaChat and try to open an individual message's context menu without using the mouse.
- Screenshots: N/A
- Logs: N/A

## Proposed solution

Implement the [listbox](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox) (https://www.w3.org/WAI/ARIA/apg/patterns/listbox/) design pattern, but with a few extras:

1. The context menu key (shift+F10 on Windows and Linux, the Applications/popup key on Windows, and also a custom implemented Shift+F10 on Mac) should open the context menu on the currently focused item.
2. Tab should move to focusable items within a message like links. If implemented correctly, this should be handled by the browser engine automatically. The tricky part is if there are multiple links in different messages, one may want to trap the tab key to move to the Attachments button instead of the next link in a different message. Right now, tab just moves through all the links that are anywhere within the current conversation.
3. Also make the divider items proper list items that can be focused, so users know when they traverse to the next/previous date.

**Comment by Jikstra** [comment]
@MarcoZehe merci beaucoup for filing all the accessibility issues! I want to focus on fixing them for the 1.16 series. The message list will receive a rewrite and in this process i want to fix this issues :)
